### DISCUSSION TITLE
Filter Points by Source / Ability to Stream Data from Car OBD2

### DISCUSSION THREAD

**Discussion by realronaldrump** [Freika/dawarich#331] [Ideas]
**Describe the bug**
This isn't a bug, but a feature that I'd love to see implemented.  I've used several sources from apps that have my location data over the last few years.  For the most part, with a little tweaking to their GPX structure, I'm able to import the data with on issue. 
 However, every now and then, an erroneous set of points are logged, creating what looks like straight lines across the map.  When a sources tens of thousands of points on a map with over 2 million + data points, it becomes very tedious trying to track down which points need to be deleted.

Additionally, I'd love to be able to use my Bouncie OBD2 device to stream location data to Dawarich.  I actually talked to their development team about it, and they said it's very doable with their open API and Trips Rest API.  Here's the API documentation:
https://docs.bouncie.dev/#api-REST_Vehicles. It would be great to be able to get historical data too.  I can provide access to the developer account I created.

Last: there's an app called ARC Timeline (partnered with ARC Recorder and ARC Mini) that serve a similar purpose to Overland and Owntracks.  However, I've actually found to be more accurate and fast with reverse geocoding, and way more user friendly to a person just starting out with looking at and working with code.  They provide daily and monthly file exports as JSON and GPX, but Dawarich doesn't accept the format of either of them.

**Version**
9.8.0

**To Reproduce**
N/A - this is a feature request, not an error

**Expected behavior**
Imports from a broader range of sources

**Screenshots**
N/A

**Logs**
N/A

**Additional context**
N/A

**Comment by Freika** [comment]
1. Filtering by source.

While it's not that hard to implement, it's also not that easy considering each app that is sending geodata to Dawarich uses a different json structure, and `device_id`/`tracker_id` is not a mandatory attribute by any means. Then there are imports that don't have this too. So filtering by source is a good idea, and it will probably be implemented at some point, there are things to consider.

2. Importing GPX

It seems to me GPX is not as standardized as I expected, and different sources can render different structure of the files. If you have a GPX file that fails to be imported, feel free to send it to me so I could have a look and maybe fix importing code.

> When a sources tens of thousands of points on a map with over 2 million + data points, it becomes very tedious trying to track down which points need to be deleted.

I get you, but you can narrow your scope by changing the timeframe, hope this helps

3. Bouncie OBD2

I can have a look, but can't promise anything, sorry. It actually would be lot better if you could create different topics in [discussions](https://github.com/Freika/dawarich/discussions) where people can upvote messages so I could see if some requests are highly upvoted so I could prioritize them.

But I have to say I love the idea of being able to track my car's position.

4. Arc Timeline

Color me interested, I installed all 3 apps to my iPhone to play with. I'll probably be able to get my own export files, but again, wouldn't hurt if you could send me anonymized GPX/JSON files to play with.

---

Resume: all your requests are valid, although I'd ask you to create a separate topic in Discussions (Ideas section) so it could be easier to see the demand.

Thank you for your interest in Dawarich!
